### PR TITLE
Submit classification

### PR DESCRIPTION
**Pull request by srallen**
Package: lib-classifier, lib-async-states

Describe your changes:
Branched off of multiple-choice-annotation, this finishes the basic back/next functionality of the task nav and submits the classification, loads the next subject, and resets the workflow steps back to start when done is clicked. 

~I'm still working on the conversion part of the classification from the store to JSON the API will accept. The mobx `toJS` method isn't quite working the way I think it should, so I'm investigating that.~

~Working on loading the next subject.~

Ready for review 

# Review Checklist

## General

- [ ] Are the tests passing locally and on Travis?
- [ ] Is the documentation up to date?
- [ ] Is the changelog updated?

## Apps

- [ ] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [ ] Does it work on mobile?
- [ ] Can you `rm -rf node_modules/ && npm install` and app works as expected?

### CODE REVIEW

**Comment by srallen** [comment]
@rogerhutchings could I get a review on this and on #194 ? I'm gonna continue on to add more tests and refactor the submit function to instead push into a queue handled by a service worker.

File: packages/lib-classifier/src/store/WorkflowStepStore.js [hunk 1]
```diff
@@ -59,14 +60,36 @@ const WorkflowStepStore = types
       addDisposer(self, workflowDisposer)
     }
 
+    function createClassificationObserver () {
+      const classificationDisposer = autorun(() => {
+        const classificationStore = getRoot(self).classifications
+        onAction(classificationStore, (call) => {
```
Comment (srallen) [inline packages/lib-classifier/src/store/WorkflowStepStore.js]: Writing tests on a separate branch and MST threw this:
`Warning: Attaching onAction listeners to non root nodes is dangerous: No events will be emitted for actions initiated higher up in the tree.`

Sooooo, I suppose I gotta rethink what's going on here.

**Comment by srallen** [comment]
I've resolved the warning when using `onAction`. Taking a look at the other MST warnings on submit...

**Comment by srallen** [comment]
Ok, I've refactored a few things in these last two commits:

- All of the defined store models have names now so this will make debugging easier. Additionally I added an option to have MST actually throw warnings so we can get the stack traces for debugging.
- I moved the call to advance after classification submission into the `SubjectStore`. At first it seemed odd to have the stores watching each other, but it's for different reasons. The `ClassificationStore` will create a new instance of a Classification when the active subject changes. The `SubjectStore` reacts to a patch of the classification store when its asyncState becomes `posting`. The `SubjectStore` doesn't need to care if the POST is actually successful (that's going to be the responsibility of the service worker), so success or failure of the async HTTP request shouldn't impact loading the next subject.
- I refactored the `SubjectStore` and removed the queue entirely. The queue was redundant with the resources model property. It created two sources of references to the Subject model and both had to be updated when the queue advanced. Instead I decided to use iterators on the resources map to get the value to set to active.
- I casted the subject props in the `SingleImageSubjectViewer` to JSON. There was an issue with the simultaneous removal of the resource while advancing and referencing it via `prevSubject.id` in the `componentDidUpdate` lifecycle method causing errors to be thrown. Casting to JSON seemed to fix it, but this doesn't *feel* like the right solution. A future refactor may be to define a getter in a view function on the model.

**Comment by srallen** [comment]
Thanks @shaunanoordin ! I should note that the additional TODOs are noted in #153 and #155. I'm going to go ahead and merge. Would you open up an issue for the layout and rendering problems for those browsers?

**Comment by shaunanoordin** [comment]
👌 Can do, I'll open those issues once I grab some screenshots.

**Pull request merged**